test/FunctionPointer/main.cpp

```diff
@@ -54,20 +59,22 @@ void runTest(void) {
     printf("sizeof(bp) = %d\r\n", ebsize);
     printf("sizeof(cp) = %d\r\n", ecsize);
 
-    bool result = false;
+    bool result = true;
     // Test checks
     {
         ebp.call();
-        result &= ebp_flag;
+        result = result && ebp_flag;
         printf("ebp_flag = %d\r\n", ebp_flag);
 
         ecp.call();
-        result &= ecp_flag;
+        result = result && ecp_flag;
         printf("ecp_flag = %d\r\n", ecp_flag);
     }
+    result = result && testInts();
```

Comment:
this pattern is going to skip the evaluation of subsequent tests if result is already false – do you want that?